src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private double threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) DEFAULT_LOAD_FACTOR * DEFAULT_CAPACITY;
+    }
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+
+        @Override
+        public int hashCode() {
+            hash = 31;
+            hash = hash * 17 + key.hashCode();
+            return hash;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K,V> newNode = new Node(key,value);
+        int index = putIndex(newNode);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        } else if (table[index].value != null) {
+            if (Objects.equals(table[index].key, key)) {
+                table[index].value = value;
+                return;
+            } else {
+                Node<K, V> currentNode = table[index];
+                while (currentNode.next != null) {
+                    currentNode = currentNode.next;
+                }
+                currentNode.next = newNode;
+                size++;
+                return;
+            }
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : table) {
+            Node<K, V> currentNode = node;
```

Comment:
do we have to check all table with for loop?
or we can use key and find node that we want to check